std/algorithm/iteration.d

```diff
@@ -4068,6 +4053,286 @@ unittest
     assert(sb == (BigInt(ulong.max/2) * 10));
 }
 
+
+/++
+Naive summation algorithm.
++/
+private F sumNaive(Range, F = sumType!Range)(Range r, F s = 0)
+{
+    foreach (x; r)
+    {
+        s += x;
+    }
+    return s;
+}
+
+///TODO
+private alias sumFast = sumNaive;
+
+/++
+$(LUCKY Pairwise summation) algorithm. Range must be a finite sliceable range.
++/
+private F sumPairwise(Range, F = Unqual!(ForeachType!Range))(Range r)
```

Comment:
I will play with @John-Colvin ideas. Lets see how we can improve summation concept.